src/components/ContentWrapper.jsx

```diff
@@ -68,30 +68,31 @@ const ContentWrapper = (props) => {
   const gem2sRunningError = backendErrors.includes(gem2sStatusKey);
   const completedGem2sSteps = backendStatus.gem2s?.completedSteps;
 
-  const changedQCFilters = useSelector((state) => state.experimentSettings.processing.meta.changedQCFilters);
+  const changedQCFilters = useSelector(
+    (state) => state.experimentSettings.processing.meta.changedQCFilters,
+  );
 
   // This is used to prevent a race condition where the page would start loading immediately
   // when the backend status was previously loaded. In that case, `backendLoading` is `false`
   // and would be set to true only in the `loadBackendStatus` action, the time between the
   // two events would allow pages to load.
   const [backendStatusRequested, setBackendStatusRequested] = useState(false);
-
   const [changesNotAppliedModalPath, setChangesNotAppliedModalPath] = useState(null);
 
-  const setUpConnection = async () => {
-    const io = await connectionPromise;
-    const cb = experimentUpdatesHandler(dispatch);
-    io.on(`ExperimentUpdates-${experimentId}`, (update) => cb(experimentId, update));
-  };
-
   useEffect(() => {
     if (!experimentId) {
       return;
     }
 
     dispatch(loadBackendStatus(experimentId));
 
-    setUpConnection();
+    (async () => {
+      const io = await connectionPromise;
+      const cb = experimentUpdatesHandler(dispatch);
+
+      io.off();
```

Comment:
why first off and then on?

Reply:
I added a comment to explain. Basically, it's in order to prevent us from accidentally getting work response messages and experiment updates from a previous experiment that the user browsed to. `io.off` effectively deletes all previous hooks that you may have created, including `WorkRespone` hooks.